cmd/crio/config.go

```diff
@@ -49,11 +49,12 @@ it later with **--config**. Global options will modify the output.`,
 		})
 		logrus.SetLevel(logrus.InfoLevel)
 
-		conf, err := criocli.GetConfigFromContext(c)
-		if err != nil {
-			return err
+		conf, ok := c.App.Metadata["config"].(*libconfig.Config)
+		if !ok {
+			return errors.New("type assertion error when accessing server config")
```

Comment:
what's the motivation for dropping this? I personally liked having this clunky piece of code in a function and not duplicated

Reply:
ah! I see

instead of duplicating this code, we should rip out this part of the function 
```
func GetConfigFromContext(c *cli.Context) (*libconfig.Config, error) {
    config, ok := c.App.Metadata["config"].(*libconfig.Config)
    if !ok {
        return nil, fmt.Errorf("type assertion error when accessing server config")
    }
}

func GetAndMergeConfigFromContext(c *cli.Context) (*libconfig.Config, error) {
    config, err := GetConfigFromContext(c)
    if err != nil {
        return nil, err
    }
    if err := mergeConfig(config, c); err != nil {
        return nil, err
    }
    return config, nil
}

Reply:
Done